[highs] upgrade to version 1.7.2

mathgeekcoder opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Fixes #40059.

mathgeekcoder committed:
Updated highs to version 1.7.2.  Removed redundant patches.

mathgeekcoder committed:
Minor fixes

mathgeekcoder committed:
highs.json sha update

WangWeiLin-MV reviewed (commented):
The patches have not been merged into the upstream. Could you include comments for removing the patch?

mathgeekcoder committed:
Added back required patches

The `fix-compiler.patch` seems to work with or without it, but it might be required on other's machines.  So I've added it back. I've also merged the `__popcnt64` fix from `fix-cmake-output.patch`.

I'm also including the highs.exe as a diagnostic tool via `vcpkg_copy_tools`.  It's quite handy, but can remove if needed.

Also added back the `fix-hconfig-path.patch`.  I had replaced it by deleting the hconfig.h file, but it was being used by a few header files.  The patch simply removes the absolute paths.

mathgeekcoder committed:
highs.json sha update

WangWeiLin-MV reviewed (approved):
The port installation tests pass with the following triplets:
* x64-linux
